### PR TITLE
Update outdated tensorflow version warning

### PR DESCRIPTION
**Pull request by apurva-koti**
TensorFlow 2.x has been supported for some time, but our version check warning text doesn't reflect this.

## What changes are proposed in this pull request?

Updates warning message

## How is this patch tested?

N/A

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/tensorflow.py [hunk 1]
```diff
@@ -829,8 +829,8 @@ def autolog(every_n_iter=100):
 
     if LooseVersion(tensorflow.__version__) < LooseVersion("1.12"):
         warnings.warn(
-            "Could not log to MLflow. Only TensorFlow versions"
-            + "1.12 <= v <= 2.0.0 are supported."
+            "Could not log to MLflow. TensorFlow versions"
```
Comment (harupy) [inline mlflow/tensorflow.py:832]:
```suggestion
            "Could not log to MLflow. TensorFlow versions "
```
nit

**Review by juntai-zheng — APPROVED** [review]
Looks good, tiny nit regarding another versioning comment we could improve on:

[2.0](https://github.com/mlflow/mlflow/blob/68c0985689994c02bca01ccb5fd458cc5019e1d5/mlflow/tensorflow.py#L368) -> we could change this to `>= 2.0` instead.

**Comment by apurva-koti** [comment]
@juntai-zheng what about `TF 2.0+`?

**Comment by apurva-koti** [comment]
Discussed offline, maintain in-file consistency and use `>=`

**Pull request merged**